packages/dds/tree/src/shared-tree-core/branch.ts

```diff
@@ -106,20 +106,30 @@ export interface SharedTreeBranchEvents<TEditor extends ChangeFamilyEditor, TCha
 	 */
 	afterChange(change: SharedTreeBranchChange<TChange>): void;
 
-	/**
-	 * Fired when a revertible change is made to this branch.
-	 */
-	newRevertible(revertible: Revertible): void;
-
 	/**
 	 * Fired when a revertible made on this branch is disposed.
 	 *
 	 * @param revertible - The revertible that was disposed.
-	 * This revertible was previously passed to the `newRevertible` event.
+	 * This revertible was previously obtained through the `commitApplied` event.
 	 * @param revision - The revision associated with the revertible that was disposed.
 	 */
 	revertibleDisposed(revertible: Revertible, revision: RevisionTag): void;
 
+	/**
+	 * Fired when:
+	 * - a local commit is applied outside of a transaction
+	 * - a local transaction is committed
+	 *
+	 * The event is not fired when:
+	 * - a local commit is applied within a transaction
+	 * - a remote commit is applied
+	 *
+	 * @param data - information about the commit that was applied
+	 * @param getRevertible - a function provided that allows users to get a revertible for the commit that was applied. If not provided,
+	 * this commit is not revertible.
+	 */
+	commitApplied(data: CommitMetadata, getRevertible?: () => Revertible): void;
```

Comment:
The spec says we provide the view that the commit is applied on but after writing this code, it doesn't seem to be necessary/helpful in anyway. Especially since the user is putting the event handler on said view in the first place. Thoughts? @nmsimons @taylorsw04